src/mlpack/methods/ann/brnn_impl.hpp

```diff
@@ -83,6 +83,15 @@ double BRNN<OutputLayerType, MergeLayerType, MergeOutputType,
     ResetParameters();
   }
 
+  if (optimizer.MaxIterations() < this->predictors.n_cols)
+  {
+    Log::Warn << "The number of iterations is less than the size of the "
+              << "dataset, the optimizer might not pass over the entire "
+              << "dataset. Please modify the number of iterations to be "
+              << "at least equal to the number of columns of your "
```

Comment:
How about printing minimum number of required iterations (Rather than saying `at be equal to number of columns`.)

Reply:
A member will provide better insight for the same.